src/clock.rs

```diff
@@ -39,7 +38,14 @@ impl Instant {
             (
                 InstantKind::Virtual { nanoseconds },
                 InstantKind::Virtual { nanoseconds: earlier },
-            ) => Duration::from_nanos(nanoseconds.saturating_sub(earlier)),
+            ) => {
+                let duration = nanoseconds.saturating_sub(earlier);
+                // It is possible for second to overflow because u64::MAX < (u128::MAX / 1e9).
+                let seconds = u64::try_from(duration.saturating_div(1_000_000_000)).unwrap();
```

Comment:
We should probably write this in a way that it can never panic, and instead saturate to `u64::MAX` seconds.

`saturating_div` is unnecessary, it only ever does anything on division by `-1`.